jax/numpy/lax_numpy.py

```diff
@@ -242,10 +242,60 @@ def _promote_args_like(op, *args):
 def _constant_like(x, const):
   return onp.array(const, dtype=_dtype(x))
 
+
+def update_numpydoc(docstr, fun, op):
+  if docstr[0] == ' ' or docstr[1] == ' ':
+    lines = docstr.split('\n')
+    for idx, line in enumerate(lines):
+      lines[idx] = line.replace('    ', '', 1)
+    docstr = '\n'.join(lines)
+
+  try:
```

Comment:
Can you please explain what is this doing? It seems to get the parameter names either from the __code__ attribute or if that does not work by finding the first occurrence of an example call for the function (by parsing "func(" followed by the first ")" ) and then trying to extract the parameter names, either positional or keyword. And then the parameter names are replaced in the "Parameters" block of the docstring (but not in the rest of the docstring)?

I am concerned that this code is very brittle, e.g., in this code "func(..., param=(1,2), ...)" you will miss the true closing parenthesis. 

Is the actual goal to align the signature of the jax variant of numpy with the official documentation?  In general, the docstring was written by humans for a certain function signature and it is unlikely that we can write solid parsing code to change the signature. E.g., the rest of the numpy docstring may have additional call examples, from which this does not remove the extra parameters.  

I think that if we just change the parameter names in jax to match those in numpy it is probably an easier aproach? For the case where numpy supports more parameters, one can see at the top of the docstring the parameters that jax's version supports. Is it really that confusing to see that the original numpy supports more arguments?

Reply:
Yeah, it's probably better to tell people to look at the top of the docstring, I tried to fix the signature but as you pointed out there are many cases where this would fail :(